### PR TITLE
soc: arm: ambiq: apollo4: add simobuck init to apollo4 init

### PR DESCRIPTION
**Pull request by RichardSWheatley**
Adds comments and simobuck init function for low power.

Signed-off-by: Richard Wheatley <richard.wheatley@ambiq.com>

### CODE REVIEW

**Comment by RichardSWheatley** [comment]
@MaureenHelm @galak @ioannisg 
We are trying to get this code checked in as soon as humanly possible. Please check the code when you have time.

**Review by tgorochowik — DISMISSED** [review]

**Comment by RichardSWheatley** [comment]
Is there any reason this is being held up?

File: soc/arm/ambiq/apollo4x/soc.c [hunk 1]
```diff
@@ -10,7 +10,14 @@
 
 static int arm_apollo4_init(void)
 {
+
+	/* Initialize for low power in the power control block */
 	am_hal_pwrctrl_low_power_init();
+
+	/* Enable SIMOBUCK for this board */
+	am_hal_pwrctrl_control(AM_HAL_PWRCTRL_CONTROL_SIMOBUCK_INIT, 0);
```
Comment (MaureenHelm) [inline soc/arm/ambiq/apollo4x/soc.c:18]: The comment suggests this is board-specific, but you're doing it unconditionally at the SoC level. Do you need to add a Kconfig to allow different boards to enable/disable the feature?

Comment (RichardSWheatley) [reply soc/arm/ambiq/apollo4x/soc.c:18]: @MaureenHelm good catch! It is chip specific and not board specific.
I will change the comment.

**Review by MaureenHelm — APPROVED** [review]

**Review by fkokosinski — APPROVED** [review]

**Pull request merged**